[mpir] build failure

**Host Environment**
- OS: Ubuntu Linux 20.04
- Compiler: GCC 9.3

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install mpir`

**Additional context**
after #12677 (removing autoconfig) mpir does not build anymore on Linux
@strega-nil

Hi @cenit 
Thanks for posting this issue.

@strega-nil 
It seems that `mpir `cannot build without `autoconfig` on Linux. Could you please help take a look?

```
../libtool: 1581: ../libtool: preserve_args+= --tag CC: not found
../libtool: 1: eval: base_compile+= gcc: not found
../libtool: 1: eval: base_compile+= -DHAVE_CONFIG_H: not found
../libtool: 1: eval: base_compile+= -I.: not found
../libtool: 1581: ../libtool: preserve_args+= --tag CC: not found
../libtool: 1: eval: base_compile+= gcc: not found
../libtool: 1: eval: base_compile+= -DHAVE_CONFIG_H: not found
../libtool: 1: eval: base_compile+= -I.: not found
../libtool: 1: eval: ../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean/mpn: not found
base_compile+= -I.././../src/3.0.0-a3f571144a.clean/mpn: not found
../libtool: 1: eval: ../libtool: 1: eval: base_compile+= -I..: not foundbase_compile+= -I..: not found

../libtool: 1: eval: ../libtool: 1: eval: base_compile+= -D__GMP_WITHIN_GMP: not foundbase_compile+= -D__GMP_WITHIN_GMP: not found

../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean: not found
../libtool: 1: eval: base_compile+= -DOPERATION_add: not found
../libtool: 1: eval: base_compile+= -Wno-implicit-function-declaration: not found
../libtool: 1: eval: base_compile+= -fPIC: not found
../libtool: 1: eval: base_compile+= -c: not found
libtool: compile: you must specify a compilation command
libtool: compile: Try `libtool --help --mode=compile' for more information.
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean: not found
../libtool: 1581: ../libtool: preserve_args+= --tag CC: not found
../libtool: 1: eval: base_compile+= -DOPERATION_dummy1: not found
../libtool: 1: eval: base_compile+= -Wno-implicit-function-declaration: not found
../libtool: 1: eval: base_compile+= -fPIC: not found
../libtool: 1: eval: base_compile+= -c: not found
libtool: compile: you must specify a compilation command
libtool: compile: Try `libtool --help --mode=compile' for more information.
make[2]: *** [add.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [dummy1.lo] Error 1
../libtool: 1: eval: base_compile+= gcc: not found
../libtool: 1: eval: base_compile+= -DHAVE_CONFIG_H: not found
../libtool: 1: eval: base_compile+= -I.: not found
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean/mpn: not found
../libtool: 1: eval: base_compile+= -I..: not found
../libtool: 1: eval: base_compile+= -D__GMP_WITHIN_GMP: not found
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean: not found
../libtool: 1: eval: base_compile+= -DOPERATION_add_1: not found
../libtool: 1: eval: base_compile+= -Wno-implicit-function-declaration: not found
../libtool: 1: eval: base_compile+= -fPIC: not found
../libtool: 1: eval: base_compile+= -c: not found
libtool: compile: you must specify a compilation command
libtool: compile: Try `libtool --help --mode=compile' for more information.
make[2]: *** [add_1.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

I had the same issue today. It worked well last Thursday.

OS: Ubuntu Linux 18.04

`vcpkg/buildtrees/mpir/build-x64-linux-dbg-err.log`
```
../libtool: 1581: ../libtool: preserve_args+= --tag CC: not found
../libtool: 1: eval: base_compile+= gcc: not found
../libtool: 1: eval: base_compile+= -DHAVE_CONFIG_H: not found
../libtool: 1: eval: base_compile+= -I.: not found
../libtool: 1581: ../libtool: preserve_args+= --tag CC: not found
../libtool: 1: eval: base_compile+= gcc: not found
../libtool: 1: eval: base_compile+= -DHAVE_CONFIG_H: not found
../libtool: 1: eval: base_compile+= -I.: not found
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean/mpn: not found
../libtool: 1: eval: base_compile+= -I..: not found
../libtool: 1: eval: base_compile+= -D__GMP_WITHIN_GMP: not found
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean/mpn: not found
../libtool: 1: eval: base_compile+= -I..: not found
../libtool: 1: eval: base_compile+= -D__GMP_WITHIN_GMP: not found
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean: not found
../libtool: 1: eval: base_compile+= -DOPERATION_add: not found
../libtool: 1: eval: base_compile+= -Wno-implicit-function-declaration: not found
../libtool: 1: eval: base_compile+= -fPIC: not found
../libtool: 1: eval: base_compile+= -c: not found
libtool: compile: you must specify a compilation command
libtool: compile: Try `libtool --help --mode=compile' for more information.
make[2]: *** [add.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
../libtool: 1: eval: base_compile+= -I.././../src/3.0.0-a3f571144a.clean: not found
../libtool: 1: eval: base_compile+= -DOPERATION_dummy1: not found
../libtool: 1: eval: base_compile+= -Wno-implicit-function-declaration: not found
../libtool: 1: eval: base_compile+= -fPIC: not found
../libtool: 1: eval: base_compile+= -c: not found
libtool: compile: you must specify a compilation command
libtool: compile: Try `libtool --help --mode=compile' for more information.
make[2]: *** [dummy1.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

--- a/ports/mpir/portfile.cmake
+++ b/ports/mpir/portfile.cmake
@@ -30,11 +30,17 @@ if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
 
     set(OPTIONS --disable-silent-rules --enable-gmpcompat --enable-cxx ${SHARED_STATIC})
 
-		string(APPEND VCPKG_C_FLAGS " -Wno-implicit-function-declaration")
-		string(APPEND VCPKG_CXX_FLAGS " -Wno-implicit-function-declaration")
+    string(APPEND VCPKG_C_FLAGS " -Wno-error")
+    string(APPEND VCPKG_CXX_FLAGS " -Wno-error")
+
+    # on Linux, autoconf is required; on macOS, it isn't
+    if(VCPKG_TARGET_IS_LINUX)
+        set(AUTOCONFIG "AUTOCONFIG")
+    endif()
     
     vcpkg_configure_make(
         SOURCE_PATH ${SOURCE_PATH}
+        ${AUTOCONFIG}
         OPTIONS ${OPTIONS}  
     )
     

--- a/ports/mpir/portfile.cmake
+++ b/ports/mpir/portfile.cmake
@@ -30,8 +30,8 @@ if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
 
     set(OPTIONS --disable-silent-rules --enable-gmpcompat --enable-cxx ${SHARED_STATIC})
 
-    string(APPEND VCPKG_C_FLAGS " -Wno-error")
-    string(APPEND VCPKG_CXX_FLAGS " -Wno-error")
+    string(APPEND VCPKG_C_FLAGS " -Wno-implicit-function-declaration")
+    string(APPEND VCPKG_CXX_FLAGS " -Wno-implicit-function-declaration")
 
     # on Linux, autoconf is required; on macOS, it isn't
     if(VCPKG_TARGET_IS_LINUX)

--- a/ports/mpir/vcpkg.json
+++ b/ports/mpir/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "mpir",
   "version-string": "3.0.0",
-  "port-version": 9,
+  "port-version": 10,
   "description": "Multiple Precision Integers and Rationals",
   "homepage": "https://github.com/wbhart/mpir",
   "supports": "!(uwp | arm)"

